lib/util/http/index.ts

```diff
@@ -28,10 +32,21 @@ export interface HttpResponse<T = string> {
   headers: any;
 }
 
+async function cloneResponse<T>(
+  promisedResponse: GotPromise<any>
+): Promise<HttpResponse<T>> {
+  const response = await promisedResponse;
+  // clone body and headers so that the cached result doesn't get accidentally mutated
+  return {
+    body: clone<T>(response.body), // TODO: is this ok if the body is *not* JSON ?
```

Comment:
```suggestion
    body: clone<T>(response.body),
```
Sure, any javascript type excerpt `undefined`